src/main/java/org/sagebionetworks/web/client/widget/table/TableEntityListGroupItem.java

```diff
@@ -95,10 +110,35 @@ public void configure(EntityHeader header, final ClickHandler clickHandler) {
 
 		icon.setType(EntityTypeUtils.getEntityType(header));
 		typeField.setText(EntityTypeUtils.getFriendlyTableTypeName(header.getType()));
+		if (EntityType.dataset.equals(EntityTypeUtils.getEntityType(header))) {
+			jsClient.getEntityBundleFromCache(header.getId(), new AsyncCallback<EntityBundle>() {
+				@Override
+				public void onFailure(Throwable caught) {
+					DisplayUtils.showErrorMessage(caught.getMessage());
+				}
+
+				@Override
+				public void onSuccess(EntityBundle result) {
+					Dataset dataset = (Dataset) result.getEntity();
+					int itemCount = dataset.getItems() == null ? 0 : dataset.getItems().size();
+					itemCountField.setText(NumberFormat.getDecimalFormat().format(itemCount));
+				}
+			});
+		}
 	}
 
 	@Override
 	public Widget asWidget() {
 		return w.asWidget();
 	}
+
+	public void setItemCountVisible(boolean visible) {
+		this.itemCountColumn.setVisible(visible);
+		if (visible) {
+			itemCountColumn.addStyleName("visible-md visible-lg");
+		} else {
+			itemCountColumn.removeStyleName("visible-md visible-lg");
+		}
```

Comment:
Need to also add/remove the responsive Bootstrap classes since the visibility itself may be overridden.